src/components/side_panel/chart/building_blocks/generic_side_panel/config_panel.ts

```diff
@@ -207,10 +308,61 @@ export class GenericChartConfigPanel extends Component<Props, SpreadsheetChildEn
       this.props.definition.dataSetsHaveTitle
     );
     if (dataSets.length) {
-      return dataSets[0].dataRange.zone.top + 1;
+      return this.dataStructure === "asColumns"
+        ? dataSets[0].dataRange.zone.left
+        : dataSets[0].dataRange.zone.top + 1;
     } else if (labelRange) {
       return labelRange.zone.top + 1;
     }
     return undefined;
   }
+
+  private transposeDataSet(
+    getters: Getters,
```

Comment:
use `this.env.model.getters` directly, no ?

Reply:
Yes, it's a relicate of the (not so) good ol' time when the `transposeDataSet` function wansn't in this file